src/librustc_trans/trans/glue.rs

```diff
@@ -459,7 +459,6 @@ fn make_drop_glue<'blk, 'tcx>(bcx: Block<'blk, 'tcx>, v0: ValueRef, t: Ty<'tcx>)
         }
         ty::ty_vec(ty, None) => tvec::make_drop_glue_unboxed(bcx, v0, ty, false),
         _ => {
-            assert!(type_is_sized(bcx.tcx(), t));
```

Comment:
this is a bit suspicious, what kind of type triggers this assertion? Is it `str`? I expect we should be dealing with ty_str in the same way as ty_vec, but it seems to be missing treatment all over this file. This is not really your problem to fix (unless you want to!). I suggest creating a ty_str branch and panic!ing in it, with a FIXME comment, leave the assertion in, and change the test to use something other than a `str`